content/en/docs/ambient/upgrade/helm-upgrade/index.md

```diff
@@ -4,96 +4,101 @@ description: Upgrading an ambient mode installation with Helm.
 
-    {{< text syntax=bash snip_id=update_helm >}}
-    $ helm repo update istio
-    {{< /text >}}
+Like sidecar mode, ambient mode upgrades make use of Tags and Revisions to allow fine-grained control over gateway upgrades, including waypoints, with simple controls for rolling back at any point. However, unlike sidecar mode, the Ambient ztunnel runs as a DaemonSet, or per-node proxy, meaning that ztunnel upgrades affect, at minimum, one entire node at a time. For this reason, we recommend using Node cordoning and draining before upgrading the ztunnel component for a given node. For the sake of simplicity, this document will demonstrate in-place upgrades of the ztunnel, which are less safe, and may involve some downtime.
 
-## In-place upgrade
+{{< warning >}}
+Ztunnel upgrades may cause downtime, and are recommended only for cordoned and drained nodes.
+{{< /warning >}}
 
-You can perform an in place upgrade of Istio in your cluster using the Helm
-upgrade workflow.
+With that in mind, this guide will expand on the following steps to upgrade Istio in ambient mode:
 
-Before upgrading Istio, it is recommended to run the `istioctl x precheck` command to make sure the upgrade is compatible with your environment.
+1. Prerequisites
+1. Install the New Control Plane
+1. Upgrade the ztunnel DaemonSet
+1. Upgrade the CNI DaemonSet
+1. Upgrade waypoints and gateways using tags
+1. Uninstall the old Control Plane
+1. Upgrade the CRDs
 
-{{< text syntax=bash snip_id=istioctl_precheck >}}
-$ istioctl x precheck
-✔ No issues found when checking the cluster. Istio is safe to install or upgrade!
-  To get started, check out <https://istio.io/latest/docs/setup/getting-started/>
-{{< /text >}}
+## Prerequisites
 
-{{< warning >}}
-[Helm does not upgrade or delete CRDs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations) when performing an upgrade. Because of this restriction, an additional step is required when upgrading Istio with Helm.
-{{< /warning >}}
+1. Organize your Tags and Revisions
 
-### Manually upgrade the CRDs and Istio base chart
+    In order to safely upgrade Ambient, your Waypoints, Gateways, and Namespaces should be using the istio.io/rev label to specify an Istio Tag which will control the version of the Istio Proxy that is running. We recommend dividing your production cluster into multiple tags to organize your upgrade. All members of a given tag will be upgraded simultaneously, so it is wise to begin your upgrade with your lowest risk applications. We do not recommend referencing revisions directly via labels for upgrades, as this process can easily result in the accidental upgrade of a large number of proxies, and is difficult to segment. To see what tags and revisions you are using in your cluster, see the section on upgrading tags.
```

Comment:
I am also fine with documenting the current tag semantics, but putting a warning around the bits that interact with MWCs - we _cannot_ carry that forward in ambient, and an ambient upgrade guide should either not recommend the use of tags, or say "this is how tags work today, in ambient they will not work this way, TBD"

Reply:
To be clear, we have already carried forward MWCs into the Ambient Beta, both for revisions and for tags.  Ambient Waypoints will not function without MWCs present in the cluster.  I am all for improving this experience, but we should not hold up documentation of current functionality in hopes of better functionality later on...

Reply:
> To be clear, we have already carried forward MWCs into the Ambient Beta, both for revisions and for tags. Ambient Waypoints will not function without MWCs present in the cluster. I am all for improving this experience, but we should not hold up documentation of current functionality in hopes of better functionality later on...

Okay, so that it's clear and I understand it, let me repeat this back to you:

- We are signed up to support the revisions feature in Ambient Beta (true)
- We are signed up to support the tags feature in Ambient Beta (true)
- We are signed up to support MWCs  in ambient beta, only indirectly, as a requirement for using tags and revisions as non-unstable features (hopefully true)

If we have to ship MWCs with ambient just support tags and revisions we probably shouldn't sign up for that - I am fine with saying "we support tags and revisions, and those are _currently_ implemented with MWCs, the implementation may change later"

Reply:
The only direct references to MWCs in this document are for listing tags and revisions.  We could, instead, instruct the user to make use of istioctl to list tags and revs, which should continue to work no matter what API we eventually come up with.  Do you think that would be better?  I have avoided that so far because typically helm users don't want to make use of istioctl...

Reply:
No, I agree with the approach here - it's ugly we make you grep thru MWCs, but we do.

Putting in a disclaimer or something that inspecting MWCs directly is subject to change and not a stable API is probably enough.